docs/blockchain/why-us-aleph.md

```diff
@@ -0,0 +1,58 @@
+- Pyrsia Issue: [pyrsia/pyrsia#0000](https://github.com/pyrsia/pyrsia/issues/0000)
+
+# Summary
+
+Why we chose to use Aleph for our consensus engine
+
+# Motivation
+
+One of Pyrsia's main design decisions is to use a blockchain for its underlying datastore. The data structure(s) of a
+blockchain are plenty challenging but nothing mind bending. However, the real challenge to blockchain is consensus;
+how do you get a bunch of strangers to trust each other and agree, when some of them might be corrupt and you could
+have any number of faults in the system?
+
+# Our core requirements #
+- No currency. We wish to avoid legal problems
+- High TPS and TTF
+- Apache 2 or compatible license
+
+Given the complexity of consensus and the data structures used to manage a blockchain, it's advisable to find existing
+code that can be leveraged; stand on the shoulders of giants, as the expression goes. However, this led us into a
+contradiction with another requirement for Pyrsia; a legal and regulatory one: No currency. Each of these existing
+libraries that implement blockchain consensus and data structures all assume there is a currency involved. After all,
+this is the biggest reason people have gravitated toward blockchain: Bitcoin and Ethereum.
+
+# Prior art
+
+## Aleph ##
+
+There was only 1 library out there that exclusively did consensus and nothing else: [Aleph](https://github.com/aleph-zero-foundation/AlephBFT).
+Also, the implementation was simple as only basic traits needed to be implemented
+1. Sign/verify
+2. I/O
+
+In addition, because `verify` and `sign` as so cleanly isolated, we are able to use proof of authority. Which is simple and gives enterprise IT admins warm fuzzy feelings.
+Finally, it's block uses very well tread data structures, that have been proven repeatedly.
+
+## Substrate ##
+Another such tech [substrate](https://substrate.io/) did much more, and had a compatible license, but had the
+assumption of a currency. We had considered setting transaction costs and gas to 0 (thus neutralising block currency), but even this was deemed too much.
+
+## Tendermint ##
+The last tech truly considered was [tendermint](https://tendermint.com/). However, their examples wouldn't compile and
+when reaching out to community, after taking extreme measures to get it working, the response was ["We don't have the
+bandwidth to maintain docs in as many languages set of docs so we're planning on removing them"](https://github.com/tendermint/tendermint/issues/7743#issuecomment-1028025629).
+In addition, a core part of their platform is _not_ license compatible.
+
+# Unresolved questions
+
+There are 2 outstanding questions:
+1. What sort of key or certificate should we use to sign things? X509, which is hierarchical or simple public/private key which are quicker and easier to deploy
+2. Should metadata be cooked right into the block and transactions themselves or should metadata be an opaque part of the payload?
```

Comment:
There's no block metadata.... it's too confusing with the "package metadata" which we have no clue about packages let alone what else we might invent

```suggestion
```

Reply:
There is no called out structure for block meta-data, but there is an understanding that there will be meta-data surrounding each artifact/transaction. It was discussed quite a lot in Atlanta and pyrsia was directly compared to another OSS project I am trying to find. 

You and I have had this discussion: Will meta data exist explicitly in our block/transaction or will users just jam data in the block payload and be left to their own devices to determine what that means and if it's meta or not?

This is still an open discussion and I am sure the team will want to have some say since they will need it. I will see about scheduling something to discuss with the team.